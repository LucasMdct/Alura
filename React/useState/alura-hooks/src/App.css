* {
  box-sizing: border-box;
}

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: top;
  font-size: calc(10px + 2vmin);
  color: white;
}


.App-header .Search input {
  margin-top: 30px;
  align-items: center;
  justify-content: center;
  width: 80%;
  padding: 0 50px;
  font-size: 20px;
}



.App-header .Search ul {
  list-style-type: none;
  text-align: left;
}

  .menu {
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: black;
  }

  .menu ul {
    list-style: none;
  }

  .menu ul li {
    line-height: 30px;
    display: inline-block;
    margin: 0 20px;
  }

  .menu ul li a {
    font-size: 18px;
    color: white;
    text-decoration: none;
  }

  .menu ul li a:hover {
    font-size: 22px;
    color: whitesmoke;
    text-decoration: none;
  }



  .App-header .Search {
    width: 80%;
    margin-top: 20px;
    background-color: rgba(0,0,0, 2);
    box-shadow: 10px 10px rgba(0,0,0, 0.8);
    
    height: 500px;
    line-height: 36px;
  }

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.App-header footer {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100px;
  background-color: black;
  margin-top: 30px;
  justify-content: center;
  align-items: center;
}

.App-header .lucas {
  text-align: center;
  padding-bottom: 20px;
  width: 100%;
  background-color: black;
  font-size: 25px;

}

.App-header footer p{
  font-size: 13px;
  margin: 10px;
  line-height: 20px;
}

