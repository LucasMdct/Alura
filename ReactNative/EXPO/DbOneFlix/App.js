import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, SafeAreaView } from 'react-native';

import Routes from './src/routes';

export default function App() {
  return (
    <SafeAreaView style={{ flex: 1 , backgroundColor: '#222'}}>
      <StatusBar style="auto" backgroundColor='#222' />
      <Routes />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    marginTop: 15,
    flex: 1,
    backgroundColor: '#222',
  },
});
