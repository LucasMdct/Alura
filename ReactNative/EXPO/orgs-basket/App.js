import { StatusBar, SafeAreaView, } from 'react-native';
import { useFonts, Montserrat_400Regular, Montserrat_700Bold,} from '@expo-google-fonts/montserrat';
import { useCallback } from 'react';
import Basket from './src/screens/baskets';
import mock from './src/mocks/basket';
import * as SplashScreen from 'expo-splash-screen';


 
export default function App() {
  const [fontLoading] = useFonts({
    "MontserratRegular": Montserrat_400Regular,
    "MontserratBold": Montserrat_700Bold,
  });
  
  const onLayoutRootView = useCallback(async () => {
    if (fontLoading) {
      await SplashScreen.hideAsync();
    }
  }, [fontLoading]);

  if (!fontLoading) {
    return null;
  }


  return (
      <SafeAreaView onLayout={onLayoutRootView} style={{ flex: 1}}  testID="splashScreen">
      <StatusBar />
        <Basket {...mock}/>
      </SafeAreaView>
  );
}
