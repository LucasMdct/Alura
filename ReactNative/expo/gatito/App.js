import React from 'react';
import 'intl';
import 'intl/locale-data/jsonp/pt-BR';

import Rotas from './src/Rotas';
import TelaPadrao from './src/componentes/TelaPadrao';


export default function App() {
 //Reactotron.log('Alura');
  return (
  <TelaPadrao>
     <Rotas />
     </TelaPadrao>
  );
}