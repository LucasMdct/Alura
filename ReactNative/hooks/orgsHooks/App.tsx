
import React from 'react';
import { SafeAreaView, StyleSheet } from 'react-native';
import Home from './src/screens/Home/index';

function App() {
  return <SafeAreaView style={styleApp.screen}>
    <Home />
  </SafeAreaView>

}

const styleApp = StyleSheet.create({
  screen: {
    flex: 1,
  }
})

export default App;
