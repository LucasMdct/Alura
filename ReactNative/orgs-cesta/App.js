import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';
import React from 'react';
import Basket from './src/screens/basket';

export default function App() {
  return (
    <View style={styles.container}>

      <Basket/>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#d3d3',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
