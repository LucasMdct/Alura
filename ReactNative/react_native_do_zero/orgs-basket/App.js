import { StatusBar, SafeAreaView , View } from 'react-native';
import { useFonts, Montserrat_400Regular, Montserrat_700Bold} from '@expo-google-fonts/montserrat';
import React from 'react';
import Basket from './src/screens/baskets/';
import mock from './src/mocks/basket';

export default function App() {
  const [fontLoading] = useFonts({
    "MontserratRegular": Montserrat_400Regular,
    "MontserratBold": Montserrat_700Bold,
  });
  
  if(!fontLoading) {
    return <View/>
  }

  return (
      <SafeAreaView>
      <StatusBar />
        <Basket {...mock}/>
      </SafeAreaView>
  );
}
