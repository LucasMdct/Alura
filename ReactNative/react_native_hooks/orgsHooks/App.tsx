
import React from 'react';
import { Text  } from 'react-native';

function App() {
  return <Text> Hello World Native Puro</Text>

}

export default App;
