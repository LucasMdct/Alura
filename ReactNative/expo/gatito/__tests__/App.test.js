
import React from "react";
import { render } from "@testing-library/react-native";
import { View } from "react-native";
import Rotas from "../src/Rotas";
import App from '../App';

jest.mock('../src/Rotas');


describe('App',() => {
    it('deve renderizar sem erros', () => {

        Rotas.mockImplementation(() => <View testID="RotasMock" />);

        const { toJSON } = render(<App/>); 
        expect((toJSON())).toMatchSnapshot();
    });
})