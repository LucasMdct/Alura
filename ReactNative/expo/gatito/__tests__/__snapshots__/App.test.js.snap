// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App deve renderizar sem erros 1`] = `
[
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#A050BE",
        "flex": 1,
      }
    }
    testID="safeAreaViewTop"
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "flex": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
      testID="keyboardView"
    >
      <View
        testID="RotasMock"
      />
    </View>
  </RCTSafeAreaView>,
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#FABE50",
        "flex": 0,
      }
    }
    testID="safeAreaViewBottom"
  />,
]
`;
